test/book/book.actions.spec.js

```diff
@@ -0,0 +1,107 @@
+const Helper = require('hubot-test-helper');
+const { expect } = require('chai');
+
+const sinon = require('sinon');
+const request = require('supertest');
+
+const { SlackActionsMultiplexer, dependenciesLocator } = require('../../utils');
+const { endpoint } = require('../../config/slackActions.config.json');
+
+const BookActions = require('../../src/book/book.actions');
+const SlackClientFactory = require('../../src/clients/SlackClientFactory');
+const RedisClientFactory = require('../../src/clients/RedisClientFactory');
+
+const slackClient = SlackClientFactory();
+const redisClient = RedisClientFactory();
+
+const tick = ms => new Promise(resolve => setTimeout(resolve, ms));
+
+describe('Books actions', () => {
+  let room = null;
+  let helper;
+
+  beforeEach(() => {
+    helper = new Helper('../factories/ScriptFactory.js');
+    room = helper.createRoom();
+
+    dependenciesLocator.forceSet(
+      'slackActionsMultiplexer', new SlackActionsMultiplexer(room.robot),
+    );
+
+    const actions = new BookActions(slackClient, redisClient);
+
+    const slackActionsMultiplexer = dependenciesLocator.get('slackActionsMultiplexer');
+    slackActionsMultiplexer.addAction(/confirm-booking/, actions.confirmBooking);
+  });
+
+  afterEach(() => {
+    sinon.restore();
+    room.destroy();
+  });
+
+  describe('confirm-booking callback', () => {
+    it('responds with 200 status when proper data was given', async () => {
+      sinon.stub(slackClient.chat, 'update').returns({ ok: true });
+      sinon.stub(redisClient, 'hget').callsFake(() => ('{ "ownerId": "QWE", "status": false }'));
```

Comment:
Why not use `JSON.stringify()` on a actual **JSON**? Also, as far as I'm concerned `redisClient.hget` returns a **promise**. I think it is important to make sure the return type is the same as in the original response.

```suggestion
      sinon.stub(redisClient, 'hget').resolves(JSON.stringify({ ownerId: 'QWE', status: false }))
```

Reply:
You are returning the plain value, where normally there is a promise returned, in multiple places.

*  https://github.com/netguru/bob-the-bot/pull/82/files#diff-acc9a22a174fd6abe6c1bcf7dfc4ef2cR46
* https://github.com/netguru/bob-the-bot/pull/82/files#diff-acc9a22a174fd6abe6c1bcf7dfc4ef2cR68
* https://github.com/netguru/bob-the-bot/pull/82/files#diff-acc9a22a174fd6abe6c1bcf7dfc4ef2cR69